### PR TITLE
Update User schema and routes

### PR DESCRIPTION
**Pull request by yash-chad**
Changed user schema: 
Added relations with other models

In app.js : 
Changed /user to /users

In infra/jwt/index.js:
Added auth middleware

In infra/encryption/index.js : 
Added encryptPassword function,
Removed comparePassword and made it inline(Since it was just one line of code)

In controllers/user.js 
Added user routes

### CODE REVIEW

**Comment by vixrant** [comment]
Related to #4.

File: controllers/user.js [hunk 1]
```diff
@@ -1,15 +1,132 @@
...
 
-router.get('/', (req, res) => res.send('Hello World!'));
-// post
-// put
-// delete
+// Endpoint:  /api/users
+//Creating a user
+router.post("/",async (req,res)=>{
+
+    const user = new User(req.body);
```
Comment (siddinc) [inline controllers/user.js]: Just make sure you apply a check to find whether an existing user with the same credentials exists in the db or not before making the new user.

File: controllers/user.js [hunk 2]
```diff
@@ -1,15 +1,132 @@
...
+    } catch (error) {
+        res.status(400).send()
+    }
+
+})
+
+
+//Endpoint: /api/users/me
+//Route to delete current user
+router.delete('/me', auth, async (req,res) => {
```
Comment (siddinc) [inline controllers/user.js]: Just make sure you apply a check to find whether the user to be deleted exists in the db or not before deleting him/her.

File: controllers/user.js [hunk 3]
```diff
@@ -1,15 +1,132 @@
...
+        res.send("User deleted")
+    } catch (error) {
+        res.status(500).send()
+    }
+})
+
+
+//Endpoint: /api/users/:id
+//Get user by id
+router.get('/:id', async (req,res) => {
```
Comment (siddinc) [inline controllers/user.js]: Just make sure you apply a check to find whether the user with the given credentials exists in the db or not before GETing the new user.

**Comment by yash-chad** [comment]
I've made the requested changes 
//Added checks before creating,deleting and updating users

**Pull request merged**